lapack/testlapack/dlagv2.go

```diff
@@ -0,0 +1,182 @@
+// Copyright ©2023 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/blas"
+	"gonum.org/v1/gonum/blas/blas64"
+)
+
+type Dlagv2er interface {
+	Dlagv2(a []float64, lda int, b []float64, ldb int, alphar, alphai, beta []float64) (csl, snl, csr, snr float64)
+}
+
+func Dlagv2Test(t *testing.T, impl Dlagv2er) {
+	rnd := rand.New(rand.NewSource(1))
+	const ldExtra = 0
+	for _, lda := range []int{2, 2 + ldExtra} {
+		for _, ldb := range []int{2, 2 + ldExtra} {
+			for aKind := 9; aKind <= 20; aKind++ {
+				for bKind := 9; bKind <= 20; bKind++ {
+					dlagv2Test(t, impl, rnd, lda, ldb, aKind, bKind)
+				}
+			}
+		}
+	}
+}
+
+func dlagv2Test(t *testing.T, impl Dlagv2er, rnd *rand.Rand, lda, ldb int, aKind, bKind int) {
+	const tol = 1e-14
+
+	a := makeDlag2TestMatrix(rnd, lda, aKind)
+	b := makeDlag2TestMatrix(rnd, ldb, bKind)
+	b.Data[b.Stride] = math.NaN() // b is lower triangular.
+	// a.Data[a.Stride] = 0
```

Comment:
I think this commented out line and the two below cause the first 3 cases in Dlagv2 to not be covered by this test. I think they should be.